From f1065396ad16201d47c0921d5857bccb8ded9fc0 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Fri, 14 Oct 2011 11:33:03 +0800
Subject: [PATCH 189/635] ENGR00159979 optimize performance to refresh UI and
 video independently.

when video playblack, UI should not be refreshed by hardware composer.
when pause video, UI refresh, but video should not refresh.
To implement that effect, distinguish the surface refresh condition
between video surface and UI surface.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index a0b6c57..1b67ad3 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -456,7 +456,9 @@ static int getActiveOuputDevice(struct hwc_context_t *ctx)
 static int hwc_set(hwc_composer_device_t *dev,
         hwc_display_t dpy,
         hwc_surface_t sur,
-        hwc_layer_list_t* list)
+        hwc_layer_list_t* list,
+        char refresh,
+        char vRefresh)
 {
 		//HWCOMPOSER_LOG_RUNTIME("==============hwc_set=1==============\n");
     struct hwc_context_t *ctx = (struct hwc_context_t *)dev;
@@ -476,12 +478,14 @@ static int hwc_set(hwc_composer_device_t *dev,
 #endif
 		//HWCOMPOSER_LOG_RUNTIME("==============hwc_set=2==============\n");
 #if 1 
-    EGLBoolean sucess = eglSwapBuffers((EGLDisplay)dpy, (EGLSurface)sur);
-    if (!sucess) {
-        return HWC_EGL_ERROR;
+    if(refresh) {
+        EGLBoolean sucess = eglSwapBuffers((EGLDisplay)dpy, (EGLSurface)sur);
+        if (!sucess) {
+            return HWC_EGL_ERROR;
+        }
     }
 #endif
-    if(list == NULL || dev == NULL) {
+    if(list == NULL || dev == NULL || !vRefresh) {
     	return 0;
     }
  		//HWCOMPOSER_LOG_RUNTIME("==============hwc_set=3==============\n");
-- 
1.8.0

